File: app/src/main/java/yagoc/UserPlayer.java
package yagoc;

import java.io.Serializable;

class UserPlayer implements Player, Serializable {
    private final String name;
    private final PieceColor set;

    UserPlayer(String name, PieceColor set) {
        this.name = name;
        this.set = set;
    }

    public PieceColor pieceColor() {
        return set;
    }

    @Override
    public Move move(Board board) {
        throw new RuntimeException(name + " cannot move without user input.");
    }

    @Override
    public PlayerType type() {
        return PlayerType.user;
    }

    public String name() {
        return name;
    }

    @Override
    public String toString() {
        return set + "\t" + name() + "\t" + type();
    }
}

File: yagoc-desktop/src/main/kotlin/jenm/yagoc/players/UserPlayer.kt
package jenm.yagoc.players

import jenm.yagoc.board.BoardView
import jenm.yagoc.board.Move
import jenm.yagoc.pieces.PieceColor
import java.io.Serializable

class UserPlayer(pieceColor: PieceColor) : Player(pieceColor, PlayerType.User), Serializable {

    override fun move(board: BoardView): Move {
        throw RuntimeException("$pieceColor requires user input.")
    }

    override fun toString() = pieceColor.toString() + "\t" + type
}